resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/Config.scala

```diff
@@ -292,6 +292,14 @@ private[spark] object Config extends Logging {
       .checkValue(value => value > 0, "Allocation batch size should be a positive integer")
       .createWithDefault(5)
 
+  val KUBERNETES_MAX_PENDING_PODS =
+    ConfigBuilder("spark.kubernetes.allocation.max.pendingPods")
```

Comment:
This introduces a new namespace, `max`, with only one child, `pendingPods`. Do you have a plan to add more? Otherwise, we need to reduce the depth like `maxPendingPods`.